enterprise-release: 26.1 Documentation (#1303)
seqeralabs/docs: platform-enterprise_docs/enterprise/upgrade.md

---
title: "Upgrade deployment"
description: "Guidance for upgrading to Platform Enterprise version 26.1"
date created: "2025-11-11"
last updated: "2026-05-06"
tags: [enterprise, update, install]
---

This page outlines the steps to upgrade your database instance and Platform Enterprise installation to version 26.1, including special considerations for upgrading from earlier versions.

:::note
- Make a backup of your Platform database prior to upgrade.

> - If you are upgrading from a version prior to 25.1, complete all intermediate major version upgrades before upgrading to 26.1.

Reviewer: ```suggestion
- If you are upgrading from a version prior to 25.1, complete all intermediate major version upgrades before upgrading to 26.1, for example from 23.1 upgrade to 24.1, then 25.1, and finally 26.1. More specific requirements are detailed below for each major version.
```

- Ensure that no pipelines are in a running state during this upgrade as active run data may be lost.
:::

## Considerations for versions prior to 24.1

- If you are upgrading from a version older than 23.4.1, update your installation to version 23.4.4 **first**, before updating to version 26.1 with the steps on this page.
- **MySQL 8 required**

>   From Seqera Enterprise version 23.4, MySQL 8 was the only supported database version. If you are running MySQL 5.6 or 5.7, you must upgrade your database to a supported MySQL version (see the 26.1 considerations below for the new database baseline) before upgrading.

Reviewer: ```suggestion
  From Seqera Enterprise version 23.4, MySQL 8 was the only supported database version. If you are running MySQL 5.6 or 5.7, you must upgrade your database to a supported MySQL version (see the [26.1 database considerations below](#database-changes) for the new baseline) before upgrading.
```

- **Redis version change and property deprecation**

>   - From Seqera Enterprise version 24.2, Redis version 6.2 or greater was required. **In 26.1, Redis 6.x is no longer supported — see the 26.1 considerations below.**

Reviewer: ```suggestion
  - From Seqera Enterprise version 24.2, Redis version 6.2 or greater was required. **In 26.1, Redis 6.x is no longer supported — see the [26.1 cache considerations](#cache-layer-changes-redis-eol-and-valkey-support) below.**
```

  - From Seqera Enterprise version 24.2, `redisson.*` configuration properties are deprecated. If you previously set `redisson.*` properties directly:
    - Replace `/redisson/*` references in AWS Parameter Store entries with `TOWER_REDIS_*`.

>     - Replace `redisson.*` references in `tower.yml` with `TOWER_REDIS_*`.

Reviewer: ```suggestion
    - Replace `/redisson/*` references in AWS Parameter Store entries with `TOWER_REDIS_*` environment variables.
    - Replace `redisson.*` references in `tower.yml` with `TOWER_REDIS_*` environment variables.
```

are these redis properties documented somewhere in the docs we could link to?

  :::note

>   Set `TOWER_REDIS_*` values directly in the `tower.yml` or AWS Parameter Store entry (for example, `TOWER_REDIS_URL: redis://...`).

Reviewer: what does this sentence mean? Arent' `TOWER_REDIS_` environment variables?

Reviewer: Could you clarify what this sentence means?

Author: This is very legacy and I'll need to do some digging around it. It looks like it's a confusing reiteration of the above steps though so it can be deleted.

### Cache layer changes: Redis EoL and Valkey support

26.1 introduces Valkey support and tightens Redis version requirements.

| Cache / version | 26.1 status | Action |
| --- | --- | --- |
| Redis 6.x | EoL upstream — no longer supported | Upgrade to Redis 7.2+ or migrate to Valkey 7+ |
| Redis 7.2 | Supported | No action |
| Redis 8.x | Supported | No action |
| Redis 9.x | Not supported | Do not upgrade Redis to version 9 |
| Valkey 7.x | Newly supported in 26.1 | Optional migration path from Redis |
| Valkey 8.x | Supported | Optional migration path from Redis |

#### Migrating from Redis to Valkey

To migrate from Redis to Valkey, update the `TOWER_REDIS_URL` connection scheme to use `valkey://` (or `valkeyss://` for TLS). The Redisson client embedded in Platform 26.1 has been upgraded to support Valkey 7 and 8 dial schemes; no further configuration is required.

:::note
Redis password and ACL configuration carry over unchanged when migrating to Valkey.
:::

### Frontend image — root user deprecation

> The frontend image running as root user is removed in 26.1. The unprivileged ("rootless") image is now the only supported frontend image. If you have not already migrated, update your `tower-svc.yml` (Kubernetes) or `docker-compose.yml` (Docker Compose) to reference the unprivileged image when downloading the new templates in the General upgrade steps below.

Reviewer: ```suggestion
The frontend image running as root user is deprecated in 26.1 in favor of the unprivileged ("rootless") image. The privileged image running as root will be removed in a future major release. If you have not already migrated, update your Kubernetes[link] or Docker Compose[link] manifests to reference the unprivileged image when downloading the new templates in the General upgrade steps below.
```

https://docs.seqera.io/changelog/seqera-enterprise/v25.3#general-warnings

See the [unprivileged frontend image documentation](https://docs.seqera.io/platform-enterprise/enterprise/platform-kubernetes#seqera-frontend-unprivileged) for security context, file system, and port differences.

### Studios — enabled on all workspaces by default

In 26.1, Studios is enabled on every workspace in your instance by default. This is a behavior change from earlier versions where Studios required explicit per-workspace enablement.

The [`TOWER_DATA_STUDIO_ALLOWED_WORKSPACES`](./configuration/overview#data-features) environment variable controls Studios availability:

| Value | Behaviour |
| --- | --- |

> | Unset (default) | Studios enabled on **all workspaces** |

Reviewer: ```suggestion
| Unset (new default) | Studios enabled on **all workspaces** |
```

| `""` (empty string) | Studios disabled on all workspaces |
| Comma-separated workspace IDs | Studios enabled only on the listed workspaces |

To preserve previous opt-in behaviour after upgrading, set `TOWER_DATA_STUDIO_ALLOWED_WORKSPACES=""` before the upgrade, or set it to a comma-separated list of workspace IDs to allow.

#### Studios container template version

> The recommended Studios container template version for 26.1 is **0.9**. If you have customized your Studios container templates, update them to the 0.9 base images during this upgrade. Templates pinned to earlier Connect versions may no longer be supported. See the [Studios migration documentation](https://docs.seqera.io/platform-enterprise/studios/managing#migrate-a-studio-from-an-earlier-container-image-template).

Reviewer: @weronikasosnowskaseqera @georgi-seqera @t0randr should the template version here be bumped?

Reviewer 2: Yess, the connect-client is now on [0.12](https://github.com/seqeralabs/connect/releases/tag/client%2Fv0.12.0) latest version so we should bump the recommended templates version to that

> ### AWS data lineage tracking via SQS (preview, AWS only)**

Reviewer: ```suggestion
### AWS data lineage tracking via SQS (preview, AWS only)
```

> 26.1 introduces a preview of AWS data lineage tracking that depends on an Amazon SQS queue. This feature is AWS-only and disabled by default. If you plan to enable it, ensure your IAM policies grant the Seqera role the relevant SQS permissions in addition to the existing [Seqera IAM permissions](../compute-envs/aws-batch#iam-user-creation).

Reviewer: Is there a docs page that we can refer to? I don't see SQS permissions in the aws batch page.

Moreover, does it work with the AWS Cloud CE too?

Author: @robnewman, is this accurate or an hallucination?

Reviewer 2: The main additional AWS permissions needed by the opt-in Lineage feature are listed and can be referred to here: https://docs.seqera.io/platform-cloud/data/data-lineage#additional-iam-permissions-required

These have been tested with both AWS Batch and AWS Cloud CE and both work 
Extra context: the main thing is that the credentials used also need to have the iam:PutRolePolicy (which we currently list as needed for both Batch and Cloud CEs) to be able to update the CE instance roles with access to the lineage storage bucket dynamically at pipeline launch time.
If the AWS resources are manually created (not via Forge) then customers need to manually add the custom permissions on the instance roles themselves (also listed in the referenced doc)

See the [unprivileged frontend image documentation](https://docs.seqera.io/platform-enterprise/enterprise/platform-kubernetes#seqera-frontend-unprivileged) for security context, file system, and port differences.
The unprivileged image is a requirement for the installation via the Helm chart[link].

### Studios — enabled on all workspaces by default

In 26.1, Studios is enabled on every workspace in your instance by default. This is a behavior change from earlier versions where Studios required explicit per-workspace enablement.

The [`TOWER_DATA_STUDIO_ALLOWED_WORKSPACES`](./configuration/overview#data-features) environment variable controls Studios availability:

| Value | Behaviour |
| --- | --- |
| Unset (new default) | Studios enabled on **all workspaces** |
| `""` (empty string) | Studios disabled on all workspaces |
| Comma-separated workspace IDs | Studios enabled only on the listed workspaces |

To preserve previous opt-in behaviour after upgrading, set `TOWER_DATA_STUDIO_ALLOWED_WORKSPACES=""` before the upgrade, or set it to a comma-separated list of workspace IDs to allow.

#### Studios container template version

> The recommended Studios container template version for 26.1 is **0.9**. If you have customized your Studios container templates, update them to the 0.9 base images during this upgrade. Templates pinned to earlier Connect versions may no longer be supported. See the [Studios migration documentation](https://docs.seqera.io/platform-enterprise/studios/managing#migrate-a-studio-from-an-earlier-container-image-template).

Author: ```suggestion
The recommended Studios container template version for 26.1 is **0.12**. If you have customized your Studios container templates, update them to the 0.12 base images during this upgrade. Templates pinned to earlier Connect versions may no longer be supported. See the [Studios migration documentation](https://docs.seqera.io/platform-enterprise/studios/managing#migrate-a-studio-from-an-earlier-container-image-template).
```

  Enterprise deployments that have customized this value previously will need to adopt the new format.

## Considerations for version 25.3

- **Secret key rotation requires backup and careful configuration**

  To configure [secret key rotation](https://docs.seqera.io/platform-enterprise/enterprise/configuration/overview#secret-key-rotation):

  - To prevent data loss, perform a backup of your Platform database and securely back up your current crypto secret key before enabling and performing key rotation.
  - All backend pods or containers for your Enterprise deployment must contain the same previous and new secret key values in their configuration.
  - All backend pods or containers must be in a ready/running state before starting the Platform cron service.

## Version 26.1 upgrade considerations

### Breaking changes

#### Audit log versions in 26.1

Seqera Platform Enterprise 26.1 introduces the audit log v2 schema as a **breaking change** for direct database consumers and custom ETL jobs.

- The legacy audit log schema remains in the `tw_audit_log` table.
- The new audit log v2 schema is written to a separate `tw_audit_log_v2` table.

> - The v2 schema is not backward-compatible with the legacy schema. Field names, structure, and pagination behavior differ.

Reviewer 3: I think we shouldn't disclose names of database tables, which are an internal detail. Maybe, as an exception, we can indicate that legacy audit logs are stored in a database table named `tw_audit_log` that is considered deprecated and subject to be deleted in the future, as it rings the bell that there's at least one customer that was relying on it exceptionally to support some workflows (related to the "Upgrade path for existing integrations" section below), but we shouldn't encourage customers to do so for v2 and not disclosing its table name is a step in that direction.

- The v2 Admin panel view and CSV export are available when `TOWER_AUDIT_LOG_V2_WRITE_MODE` is set to `dual` or `v2`.

Use `TOWER_AUDIT_LOG_V2_WRITE_MODE` to control how new audit events are written:

- `v1`: Write new events to the legacy `tw_audit_log` table only. This is the default in 26.1.
- `dual`: Write new events to both `tw_audit_log` and `tw_audit_log_v2`. This is the recommended 26.1 migration mode if you need to validate the v2 schema while keeping existing v1 integrations unchanged.

> - `v2`: Write new events to `tw_audit_log_v2` only.

Reviewer 3: Similarly to a previous comment, better not to disclose database table names: maybe we can just say "v1 table" or "v2 table" (referring to the respective tables we can see in the GUI), or "v1 schema" and "v2 schema" (as it is used elsewhere), but the database table names are implementation details not meant to be disclosed.

> - `v2`: Write new events to `tw_audit_log_v2` only.

Reviewer 3: Might be worth indicating that `v2` is expected to be the default value in future versions (in fact, we might delete the toggle altogether).

Author: I think we can leave that out for now unless we know exactly which version the change applies to.

Reviewer 3: Fair.

We can do this:
- Remove `v1` from the list, as users shouldn't get the impression that they can choose to keep using v1.
- Indicate that `dual` is the default value: they can choose `v2` mode though.

##### Upgrade path for existing integrations

If you have existing scripts, exports, or ETL processes that read from the legacy audit log schema, plan the 26.1 upgrade in two stages:

> 1. Upgrade to 26.1 and set `TOWER_AUDIT_LOG_V2_WRITE_MODE=dual`.

Reviewer 3: The `dual` value is already set by default in Enterprise instances: no need to set it explicitly.

> 2. Validate your integrations against the v2 schema while your existing v1 readers continue to work from the legacy table.

Reviewer 3: We shouldn't encourage customers to read from the database, but the `schema` or `table` keywords might suggest so. Users are expected to consume the audit logs through interfaces (CSV export, API).

In the 26.1 migration plan, dual-write is transitional. Plan for 26.2 to make v2 the only write-side schema, while the legacy v1 data remains available for reads as long as your retention policy still covers the required historical period.

#### Configuration changes

- `TOWER_AUDIT_LOG_V2_ENABLED` and `TOWER_AUDIT_LOG_V2_WRITE_MODE` added as configuration options.

  - `TOWER_AUDIT_LOG_V2_WRITE_MODE`: Turns on the v2 Audit Log for parallel writes with v1 Audit Log.

>   - `TOWER_AUDIT_LOG_V2_ENABLED`: Turns on or off the v2 Audit Log view from the Admin Panel.

Reviewer 3: `TOWER_AUDIT_LOG_V2_ENABLED` does not exist and `TOWER_AUDIT_LOG_V2_WRITE_MODE` is already set to the expected value, so maybe this section could be removed.

- Run the `/migrate-db.sh` script provided in the `migrate-db` container. This will migrate the database schema.
- Deploy Seqera following your usual procedures.

## Nextflow launcher image

> If you host your nf-launcher container image on a private image registry, copy the [nf-launcher image](https://quay.io/seqeralabs/nf-launcher:j17-25.10.x) *(confirm exact tag at release cut)* to your private registry. Then update your `tower.env` with the launch container environment variable:

Reviewer: ```suggestion
If you host your nf-launcher container image on a private image registry, copy the [nf-launcher image](https://quay.io/seqeralabs/nf-launcher:j17-25.10.x) *(TODO: confirm exact tag at release cut)* to your private registry. Then update your `tower.env` with the launch container environment variable:
```
marking this with a TODO
let's keep this comment open for visibility, I'll also mark my review as needing changes

> If you host your nf-launcher container image on a private image registry, copy the [nf-launcher image](https://quay.io/seqeralabs/nf-launcher:j17-25.10.x) *(confirm exact tag at release cut)* to your private registry. Then update your `tower.env` with the launch container environment variable:

Reviewer: ```suggestion
If you host your nf-launcher container image on a private image registry, copy the [nf-launcher image](https://quay.io/seqeralabs/nf-launcher:j17-25.10.x) *(confirm exact tag at release cut)* to your private registry. Then set the launch container environment variable on your backend environment:
```

```
TOWER_LAUNCH_CONTAINER=<FULL_PATH_TO_YOUR_PRIVATE_IMAGE>
```

:::caution

> If you're using AWS Batch, you will need to [configure a custom job definition](https://docs.seqera.io/platform-enterprise/enterprise/advanced-topics/custom-launch-container) and populate the `TOWER_LAUNCH_CONTAINER` with the job definition name instead.

Reviewer: why the absolute link which may break when we'll rename vNext to v26 and version it?

Author: That's Claude - I have changed the prompt to force relative links but it does get confused with the changelog which needs absolute links. Probably need two separate skills.

- Run the `/migrate-db.sh` script provided in the `migrate-db` container. This will migrate the database schema.
- Deploy Seqera following your usual procedures.

## Nextflow launcher image

If you host your nf-launcher container image on a private image registry, copy the [nf-launcher image](https://quay.io/seqeralabs/nf-launcher:j17-25.10.x) *(confirm exact tag at release cut)* to your private registry. Then set the launch container environment variable on your backend environment:

```
TOWER_LAUNCH_CONTAINER=<FULL_PATH_TO_YOUR_PRIVATE_IMAGE>
```

:::caution

> If you're using AWS Batch, you will need to [configure a custom job definition](https://docs.seqera.io/platform-enterprise/enterprise/advanced-topics/custom-launch-container) and populate the `TOWER_LAUNCH_CONTAINER` with the job definition name instead.

Author: ```suggestion
If you're using AWS Batch, you will need to [configure a custom job definition](../enterprise/advanced-topics/custom-launch-container) and populate the `TOWER_LAUNCH_CONTAINER` with the job definition name instead.
```
